vmicore/src/vmi/InterruptEvent.cpp

```diff
@@ -125,16 +126,34 @@ void InterruptEvent::storeOriginalValue()
         throw VmiException(fmt::format(
             "{}: InterruptEvent originalValue @ {} is already an INT3 breakpoint.", __func__, targetPAString));
     }
+#elif defined(ARM64)
+    originalValue = vmiInterface->read32PA(targetPA);
+    logger->debug("Save original value",
+                  {logfield::create("targetPA", targetPAString),
+                   logfield::create("originalValue", fmt::format("{:#x}", originalValue))});
+    if (originalValue == BRK64_BREAKPOINT)
+    {
+        throw VmiException(fmt::format(
+            "{}: InterruptEvent originalValue @ {} is already an BRK64 breakpoint.", __func__, targetPAString));
+    }
+#endif
 }
 
-event_response_t InterruptEvent::_defaultInterruptCallback(__attribute__((unused)) vmi_instance_t vmi,
-                                                           vmi_event_t* event)
+event_response_t InterruptEvent::_defaultInterruptCallback(vmi_instance_t vmi, vmi_event_t* event)
 {
     auto eventResponse = VMI_EVENT_RESPONSE_NONE;
     try
     {
+#if defined(X86_64)
+        (void)(vmi);
         auto eventPA =
             (event->interrupt_event.gfn << PagingDefinitions::numberOfPageIndexBits) + event->interrupt_event.offset;
+#elif defined(ARM64)
+        addr_t eventPA;
+        if (VMI_SUCCESS !=
+            vmi_pagetable_lookup(vmi, event->arm_regs->ttbr1 & VMI_BIT_MASK(12, 47), event->arm_regs->pc, &eventPA))
```

Comment:
I would prefer making the bitmask a named constant so it becomes clear what it is supposed to accomplish